### PR TITLE
PHPLIB-1335 Add tests on Element Query Operators

### PR DESCRIPTION
**Pull request by GromNaN**
Fix [PHPLIB-1335](https://jira.mongodb.org/browse/PHPLIB-1335)

https://www.mongodb.com/docs/manual/reference/operator/query/#element

- `$exists` (skipped the last redundant test)
- `$type`

### CODE REVIEW

File: tests/Builder/Query/TypeOperatorTest.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Test $type query
+ */
+class TypeOperatorTest extends PipelineTestCase
+{
+    public function testQueryingByArrayType(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::match(
+                zipCode: Query::type('array'),
```
Comment (alcaeus) [inline tests/Builder/Query/TypeOperatorTest.php:21]: As an idea for the future: we could provide an enum for the various BSON types supported by this operator here.

File: tests/Builder/Query/TypeOperatorTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        );
+
+        $this->assertSamePipeline(Pipelines::TypeQueryingByMinKeyAndMaxKey, $pipeline);
+    }
+
+    public function testQueryingByMultipleDataType(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::match(
+                zipCode: Query::type([2, 1]),
```
Comment (alcaeus) [inline tests/Builder/Query/TypeOperatorTest.php]: Would it make sense to make the type argument variadic?
```suggestion
                zipCode: Query::type(2, 1),
```

**Review by alcaeus — APPROVED** [review]
A couple of suggested improvements for the `$type` operator, but I think we can defer those to a separate ticket.

File: tests/Builder/Query/TypeOperatorTest.php (hunk 1 above)
Comment (GromNaN) [reply tests/Builder/Query/TypeOperatorTest.php:21]: Ticket created: https://jira.mongodb.org/browse/PHPLIB-1363

File: tests/Builder/Query/TypeOperatorTest.php (hunk 2 above)
Comment (GromNaN) [reply tests/Builder/Query/TypeOperatorTest.php]: Updated. The array for will always be used, even for a single value.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**